File: src/main/java/com/destino/projeto_destino/util/usuario/Telefone/TelefoneConverter.java
package com.destino.projeto_destino.util.usuario.Telefone;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter(autoApply = true)
public class TelefoneConverter implements AttributeConverter<Telefone, String> {

    @Override
    public String convertToDatabaseColumn(Telefone attribute) {
        // Salva a string pura no DB
        return attribute == null ? null : attribute.getValorPuro();
    }

    @Override
    public Telefone convertToEntityAttribute(String dbData) {
        // Carrega o dado puro do DB e cria o Objeto de Valor, acionando a validação.
        return dbData == null ? null : new Telefone(dbData);
    }
}

File: src/main/kotlin/com/fatec/destino/util/model/usuario/telefone/TelefoneConverter.kt
package com.fatec.destino.util.model.usuario.telefone

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter(autoApply = true)
class TelefoneConverter : AttributeConverter<Telefone?, String?> {
    override fun convertToDatabaseColumn(attribute: Telefone?): String? {
        // Salva a string pura no DB
        return attribute?.valorPuro
    }

    override fun convertToEntityAttribute(dbData: String?): Telefone? {
        // Carrega o dado puro do DB e cria o Objeto de Valor, acionando a validação.
        return if (dbData == null) null else Telefone(dbData)
    }
}